lib/discourse_updates.rb

```diff
@@ -186,6 +186,21 @@ def bump_last_viewed_feature_date(user_id, feature_date)
       Discourse.redis.hset(last_viewed_feature_dates_for_users_key, user_id.to_s, feature_date)
     end
 
+    def clean_state
```

Comment:
Im nitpicking here and it's nothing that needs to be done, but this is an interesting case. Can you think of a way to ensure someone adding a new key in `DiscourseUpdates` will also think to add it here?

As I said, feel free to ignore it, it's just a design question which can be interesting.

Reply:
Interesting question. I guess one thing we could do that would mostly work is to make the `clean_state` method look at private methods of  `DiscourseUpdates` and select all methods that end with `_redis_key` and then call them and pass their returned values to `Discourse.redis.del`. Some keys would need special handling like `missing_versions_key_prefix` and `new_features_last_seen_key`, but this is one idea that could work.